test/e2e/windows/version.go

```diff
@@ -0,0 +1,16 @@
+package windows
+
+import "k8s.io/apimachinery/pkg/util/sets"
+
+type ServerVersion string
+
+const (
+	// Server2019 represent Windows Server 2019
+	Server2019 ServerVersion = "2019"
+	// Server2022 represent Windows Server 2022
+	Server2022 ServerVersion = "2022"
+)
+
+// SupportedVersions are the Windows Server versions supported by the e2e test.
+// "" implies the default which is Server2022
+var SupportedVersions = sets.New[ServerVersion](Server2019, Server2022, "")
```

Comment:
I don't personally agree with introducing this package/type here.
In this case it doesn't solve any problem that a map doesn't already. Looking at the [package],(https://pkg.go.dev/k8s.io/apimachinery/pkg/util/sets#Set) I understand using it if you need to do anything related to set theory, but in my opinion its better to use the built in solution here,

Reply:
Why would I use a map? What do you mean by the built in solution? Do you mean a regular list?

Not sure why you think the set package should be used only if we are doing something related to set theory. This is used heavily upstream in similar scenarios. It just made the code look more elegant and we were already vendoring the library. But I can remove it if you insist.

Reply:
While I don't agree with your view that using the set package is only for set theory, I have gone ahead and switch to using a list. If you want a map, I am going to insist on using a set 😃

Reply:
I was referring to using this pattern that we use elsewhere through the codebase
```
versions := map[string]struct{}{"2019":struct{}}
if _, exists := versions["2019"]; exists {
   ....
}
```
I spent some time looking into this more and I guess its a less common go pattern than I thought :sweat_smile:

Reply:
Example in our code: https://github.com/openshift/windows-machine-config-operator/blob/master/controllers/controllers.go#L254